driver-sync/src/test/functional/com/mongodb/client/JsonPoweredCrudTestHelper.java

```diff
@@ -751,4 +810,8 @@ WriteConcern getWriteConcern(final BsonDocument arguments) {
     ReadConcern getReadConcern(final BsonDocument arguments) {
         return new ReadConcern(ReadConcernLevel.fromString(arguments.getDocument("readConcern").getString("level").getValue()));
     }
+
+    boolean isSharded() {
+        return database.runCommand(Document.parse("{isMaster: 1}")).get("msg", "").equals("isdbgrid");
```

Comment:
Can't use the helper from core as that requires a local mongod running on the default port. Which breaks the test when running against embedded as there is no local mongod on 127.0.0.1